From 3935fbc79ecfc101b8239d9111614dfe85387ec8 Mon Sep 17 00:00:00 2001
From: Rick Farrington <Ricardo.Farrington@cavium.com>
Date: Fri, 22 Nov 2019 00:32:38 -0500
Subject: [PATCH 1012/1239] octeontx: pcie-console: rework commit 5c90bf5; fix
 nexus hdr version

This is a rework of commit 5c90bf5 so as to be consistent with
endian usage.

Change-Id: I58c088964fad16d76e3344f23119ee085c5a5e7c
Signed-off-by: Rick Farrington <Ricardo.Farrington@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/19119
Reviewed-by: Aaron Williams <awilliams@marvell.com>
Reviewed-by: Derek Chickles <Derek.Chickles@cavium.com>
Tested-by: Chandrakala Chavva <cchavva@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/19369
Tested-by: Suneel Garapati <sgarapati@caviumnetworks.com>
Reviewed-by: Suneel Garapati <sgarapati@caviumnetworks.com>
---
 drivers/serial/serial_octeontx_pcie_console.c | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/drivers/serial/serial_octeontx_pcie_console.c b/drivers/serial/serial_octeontx_pcie_console.c
index 48665eced8..a358a79c1d 100644
--- a/drivers/serial/serial_octeontx_pcie_console.c
+++ b/drivers/serial/serial_octeontx_pcie_console.c
@@ -918,12 +918,11 @@ int octeontx_pcie_console_init_nexus(struct udevice *dev)
 	}
 
 	new_hi = cpu_to_le64(OCTEONTX_PCIE_CONSOLE_NEXUS_MAGIC);
-	new_lo = ((u64)OCTEONTX_PCIE_CONSOLE_MAJOR << 56) |
-		  (u64)OCTEONTX_PCIE_CONSOLE_MINOR << 48 |
-		  (u64)0 /*i.e. flags */ << 40 |
-		  (u64)num_consoles << 32;
-	/* these struct fields are declared as bytes, which corresponds to BE */
-	new_lo = cpu_to_be64(new_lo);
+	new_lo = ((u64)OCTEONTX_PCIE_CONSOLE_MAJOR << 0) |
+		  (u64)OCTEONTX_PCIE_CONSOLE_MINOR << 8 |
+		  (u64)0 /*i.e. flags */ << 16 |
+		  (u64)num_consoles << 24;
+	new_lo = cpu_to_le64(new_lo);
 
 	debug("%s: Writing %#llx %#llx to %p\n", __func__, new_hi, new_lo, pcd);
 	/* Now fill in the header atomically */
-- 
2.29.0

